api/server/runner.go

```diff
@@ -29,42 +28,37 @@ type runnerResponse struct {
 
 func (s *Server) handleSpecial(c *gin.Context) {
 	ctx := c.MustGet("ctx").(context.Context)
-	log := common.Logger(ctx)
+	ctx, log := fcommon.LoggerWithStack(ctx, "server.handleSpecial")
+	log.Infoln("handleSpecial called")
 
-	ctx = context.WithValue(ctx, api.AppName, "")
-	c.Set(api.AppName, "")
-	ctx = context.WithValue(ctx, api.Path, c.Request.URL.Path)
-	c.Set(api.Path, c.Request.URL.Path)
+	// middleware should pass this or a new one along to Next()
+	app := &models.App{}
 
-	ctx, err := s.UseSpecialHandlers(ctx, c.Request, c.Writer)
-	if err == ErrNoSpecialHandlerFound {
-		log.WithError(err).Errorln("Not special handler found")
-		c.JSON(http.StatusNotFound, http.StatusText(http.StatusNotFound))
-		return
-	} else if err != nil {
-		log.WithError(err).Errorln("Error using special handler!")
-		c.JSON(http.StatusInternalServerError, simpleError(errors.New("Failed to run function")))
-		return
+	fctx := &middlewareContextImpl{
+		Context: ctx,
 	}
+	fctx.app = app
+	// fctx.index = -1
+	fctx.ginContext = c
+	fctx.middlewares = s.runMiddlewares
```

Comment:
Okay. I was thinking this was a middleware for the runner. Why are we using it on the special handler?

Did I just get the whole intention wrong?